### PR TITLE
added prefix to discovery project and collection, fixed watson discovery variable names

### PR DESCRIPTION
**Pull request by brendankellyibm**
### Description

- Added prefix to discovery project and collection names, if the prefix variable is specified
- Fixed discovery variable names from WatsonX to Watson
- Future PR will handle custom names for both Discovery and Assistant

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by brendankellyibm** [comment]
@ocofaigh @vburckhardt struggling a bit with the upgrade test again, can you check to see what needs to be done to fix it?

File: solutions/banking/main.tf [hunk 1]
```diff
@@ -1,7 +1,9 @@
 locals {
-  watsonx_assistant_url = "https://api.${var.watson_assistant_region}.assistant.watson.cloud.ibm.com/instances/${var.watson_assistant_instance_id}"
-  watsonx_discovery_url = "//api.${var.watson_discovery_region}.discovery.watson.cloud.ibm.com/instances/${var.watson_discovery_instance_id}"
-  sensitive_tokendata   = sensitive(data.ibm_iam_auth_token.tokendata.iam_access_token)
+  watsonx_assistant_url            = "https://api.${var.watson_assistant_region}.assistant.watson.cloud.ibm.com/instances/${var.watson_assistant_instance_id}"
+  watson_discovery_url             = "//api.${var.watson_discovery_region}.discovery.watson.cloud.ibm.com/instances/${var.watson_discovery_instance_id}"
+  watson_discovery_project_name    = var.prefix != "" ? "${var.prefix}-gen-ai-rag-sample-app-project" : "gen-ai-rag-sample-app-project"
```
Comment (ocofaigh) [inline solutions/banking/main.tf:4]: `var.prefix != ""` should actually be `var.prefix != null` - `null` and empty strings are two completely different things in terraform. I guess you can also check for empty string here too, but its unlikley someone would enter an empty string.

File: solutions/banking/main.tf [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
+  watson_discovery_collection_name = var.prefix != "" ? "${var.prefix}-gen-ai-rag-sample-app-data" : "gen-ai-rag-sample-app-data"
```
Comment (ocofaigh) [inline solutions/banking/main.tf:5]: same as above comment

**Comment by ocofaigh** [comment]
@brendankellyibm Did you see my comments above?

**Comment by brendankellyibm** [comment]
@ocofaigh yes - I don't foresee this breaking existing deployments, as the prefix is optional. I think this is good to go, unless you have more concerns.

**Comment by ocofaigh** [comment]
@brendankellyibm see https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/pull/113#discussion_r1675960572 and https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/pull/113#discussion_r1675960931

**Comment by brendankellyibm** [comment]
> @brendankellyibm see [#113 (comment)](https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/pull/113#discussion_r1675960572) and [#113 (comment)](https://github.com/terraform-ibm-modules/terraform-ibm-rag-sample-da/pull/113#discussion_r1675960931)

Ah sorry, didn't see _those_ comments. Will fix.

File: solutions/banking/main.tf (hunk 2 above)
Comment (brendankellyibm) [reply solutions/banking/main.tf]: Updated as above

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**